pkg/azuremachinepool/validate_azuremachinepool_create.go

```diff
@@ -0,0 +1,88 @@
+package azuremachinepool
+
+import (
+	"context"
+	"fmt"
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/v2/pkg/apis/infrastructure/v1alpha2"
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+	restclient "k8s.io/client-go/rest"
+	expcapzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	apiv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
```

Comment:
We are using v1alpha3

Reply:
Maybe this is why it's not working fuck me

Reply:
bumped